### PR TITLE
REFACTOR-#2739: io tests refactoring

### LINKED ISSUE #2739: IO tests refactoring
IO tests common functions can be moved into `utils` module and fixtures can be moved into `modin/conftest.py` file where they can accessed from `modin/experimental` directories.

### PR DESCRIPTION
**Pull request by amyskov**
Signed-off-by: Alexander Myskov <alexander.myskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2739  
- [x] tests added and passing

### CODE REVIEW

File: modin/pandas/test/utils.py [hunk 1]
```diff
@@ -1071,3 +1086,138 @@ def wrapped_function(self, *args, **kwargs):
...
+                    lambda x: f"{x:,f}".replace(",", thousands_separator)
+                )
+            filename = (
+                f"{filename}.{COMP_TO_EXT[compression]}"
+                if compression != "infer"
+                else filename
+            )
+            # create test data dir if it is not exists yet
+            if not os.path.exists(IO_OPS_DATA_DIR):
+                os.mkdir(IO_OPS_DATA_DIR)
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:1164]: Why is this folder created for only `csv` files? What about other types, in instance `parquet`?

File: modin/pandas/test/utils.py [hunk 2]
```diff
@@ -39,9 +39,21 @@
     "Big": (2 ** 7, 2 ** 12),
 }
 
+# dataset sizes for IO operations
+DATASET_SIZE_DICT_IO = {
+    "Small": 64,
+    "Normal": 2000,
+    "Big": 20000,
+}
+
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:48]: You also have a similar variable in `modin/pandas/test/test_io.py` - `DATASET_SIZE_DICT`. Can we reuse it?

File: modin/pandas/test/utils.py (hunk 1 above)
Comment (amyskov) [reply modin/pandas/test/utils.py]: Right, this directory should be created for all io tests, so directory creation was moved on the top of `conftest.py` file.

File: modin/pandas/test/utils.py (hunk 2 above)
Comment (amyskov) [reply modin/pandas/test/utils.py]: Yes, i think we can use number of rows from `DATASET_SIZE_DICT`, default number of rows is larger, but tests execution time ingresses negligible.

**Review by anmyachev — APPROVED** [review]

**Pull request merged**